File: app/src/main/java/org/mozilla/focus/tabs/TabModel.java
/* -*- Mode: Java; c-basic-offset: 4; tab-width: 20; indent-tabs-mode: nil; -*-
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package org.mozilla.focus.tabs;

import android.graphics.Bitmap;
import android.os.Bundle;

/**
 * A model class to describe a tab, and these fields are supposed to be stored in persistent storage
 * for restoring tabs if app is killed.
 */
public class TabModel {

    /* package */ String title;

    /* package */ String url;

    /**
     * ViewState for this Tab. Usually to fill by WebView.saveViewState(Bundle)
     */
    /* package */ Bundle webViewState;

    /**
     * Thumbnail for this tab for previewing.
     */
    /* package */ Bitmap thumbnail;
}

File: components/feature/tabs/src/main/java/org/mozilla/rocket/tabs/TabModel.kt
package org.mozilla.rocket.tabs

import android.graphics.Bitmap
import android.os.Bundle

data class TabModel(
    val id: String,
    var parentId: String?,
    var title: String?,
    var url: String?
) {

    /**
     * Favicon bitmap for tab tray item.
     */
    var favicon: Bitmap? = null

    /**
     * Thumbnail bitmap for tab previewing.
     */
    var thumbnail: Bitmap? = null

    /**
     * ViewState for this Tab. Usually to fill by WebView.saveViewState(Bundle)
     * Set it as @Ignore to avoid storing this field into database.
     * It will be serialized to a file and save the uri path into webViewStateUri field.
     */
    var webViewState: Bundle? = null

    fun isValid(): Boolean {
        return id.isNotBlank() && (url?.isNotBlank() ?: false)
    }
}
